### ISSUE TITLE
Liquid: Nested if statements

### ISSUE THREAD

**Issue by macdonaldr93** [prettydiff/prettydiff#284]
It looks like nested `if` statements are causing issues. It associates `else` to the first `if` rather than the second. This is just a snippet of the larger file.

## Source

```
{% if cat != tag and cat_item == cat %}
  {% comment %}
    Curabitur vestibulum aliquam leo. Nam commodo suscipit quam
  {% endcomment %}
  {% if current_tags contains tag %}
    <li class="lorem-filter active-filter" data-group="{{ cat_item }}" data-handle="{{ tag | handle }}">{{ tag | remove_first: cat_item | remove_first: '_' | link_to_remove_tag: tag }}</li>
  {% else %}
    <li class="lorem-filter" data-group="{{ cat_item }}" data-handle="{{ tag | handle }}">{{ tag | remove_first: cat_item | remove_first: '_' | link_to_add_tag: tag }}</li>
  {% endif %}
{% endif %}
```

## Output

```
{% if cat != tag and cat_item == cat %}
    {% comment %}
    Curabitur vestibulum aliquam leo. Nam commodo suscipit quam
    {% endcomment %}
    {% if current_tags contains tag %}
    <li
        class="lorem-filter active-filter"
        data-group="{{ cat_item }}"
        data-handle="{{ tag | handle }}">{{ tag | remove_first: cat_item | remove_first: '_' | link_to_remove_tag: tag }}</li>
{% else %}
    <li
        class="lorem-filter"
        data-group="{{ cat_item }}"
        data-handle="{{ tag | handle }}">{{ tag | remove_first: cat_item | remove_first: '_' | link_to_add_tag: tag }}</li>
{% endif %}
{% endif %}
```

**Comment by prettydiff** [comment]
Try it out at http://prettydiff.com/ignore/testlocation/index.xhtml?m=beautify

**Comment by macdonaldr93** [comment]
Yup, detected the if statements properly! Not sure if you want me to make a separate issue for this, if yes, just let me know... comment formatting is still a little off in the output and `li` element attributes are outputting strangely. I'm not sure if that `li` one is related to a setting though.

## Output

```
{% if cat != tag and cat_item == cat %}
    {% comment %}
    Curabitur vestibulum aliquam leo. Nam commodo suscipit quam
  {% endcomment %}
    {% if current_tags contains tag %}
        <li
            class="lorem-filter active-filter"
            data-group="{{ cat_item }}"
            data-handle="{{ tag | handle }}">{{ tag | remove_first: cat_item | remove_first: '_' | link_to_remove_tag: tag }}</li>
    {% else %}
        <li
            class="lorem-filter"
            data-group="{{ cat_item }}"
            data-handle="{{ tag | handle }}">{{ tag | remove_first: cat_item | remove_first: '_' | link_to_add_tag: tag }}</li>
    {% endif %}
{% endif %}
```

## Issues
For the issues mentioned above, this is what I would have expected for output:

```
{% if cat != tag and cat_item == cat %}
    {% comment %}
        Curabitur vestibulum aliquam leo. Nam commodo suscipit quam
    {% endcomment %}
    {% if current_tags contains tag %}
        <li class="lorem-filter active-filter" data-group="{{ cat_item }}" data-handle="{{ tag | handle }}">{{ tag | remove_first: cat_item | remove_first: '_' | link_to_remove_tag: tag }}</li>
    {% else %}
        <li class="lorem-filter" data-group="{{ cat_item }}" data-handle="{{ tag | handle }}">{{ tag | remove_first: cat_item | remove_first: '_' | link_to_add_tag: tag }}</li>
    {% endif %}
{% endif %}
```

**Comment by prettydiff** [comment]
The comments are preserved verbatim exactly like an HTML.  This includes absolutely all whitespace and the end comment tag.  Could you be more specific about how the attributes are off?

**Comment by macdonaldr93** [comment]
I would expect the `li` attributes to remain on the same line as the `li` element. After some testing, I'm curious, are they breaking to their own lines due to the length of this line?

**Comment by prettydiff** [comment]
The attributes get word wrap if the length of the current line is greater than the value supplied in option `wrap`.  The default value is `80`.  You can supply a value of `0` to disable wrapping

**Comment by macdonaldr93** [comment]
Okay cool, that makes sense. I had a feeling it was just a setting that I would need to configure. This one looks great and I think it's good to close. I'll provide you the info required for the tags on the other issue https://github.com/prettydiff/prettydiff/issues/285.

**Comment by prettydiff** [comment]
Released with version v1.16.28.
